core/controllers/topic_editor.py

```diff
@@ -84,6 +88,48 @@ def post(self, topic_id):
         })
 
 
+class TopicEditorQuestionHandler(base.BaseHandler):
+    """Manages the creation of a question and receiving of all question
+    summaries for display in topic editor page.
+    """
+
+    @acl_decorators.can_edit_topic
+    def get(self, topic_id):
+        """Handles GET requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+        topic_domain.Topic.require_valid_topic_id(topic_id)
+
+        topic = topic_services.get_topic_by_id(topic_id)
+        skill_ids = topic.get_all_skill_ids()
+
+        question_summaries = (
+            question_services.get_question_summaries_linked_to_skills(skill_ids)
+        )
+        question_summary_dicts = [
+            summary.to_dict() for summary in question_summaries]
+
+        self.values.update({
+            'question_summary_dicts': question_summary_dicts
+        })
+        self.render_json(self.values)
+
+    @acl_decorators.can_edit_topic
+    def post(self, topic_id):
+        """Handles POST requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+        topic_domain.Topic.require_valid_topic_id(topic_id)
+
+        new_question_id = question_services.get_new_question_id()
+        question = question_domain.Question.create_default_question(
```

Comment:
I still don't understand this part. We shouldn't be adding a question until the question body is completed. (So, a creator fills out a form and submits; until they submit, the question is not a question.)

The frontend should probably have some sort of default question data dict but no adding/saving should happen until the topic manager or admin submits the form.

Reply:
Oh, ok, yeah. I thought isolated default questions wouldn't be a problem, but come to think of it, it might become too much. Will change this.

Reply:
I will add the question creation function in question_editor for now and remove this, then in the frontend PR, I will add create default in the forntned.